### PR TITLE
Simplify examples by updating to new default loop

### PR DESCRIPTION
**Pull request by SimonFrings**
This changeset simplifies usage by using the new [default loop](https://github.com/reactphp/event-loop#loop).
Builds on top of https://github.com/reactphp/event-loop/pull/229, reactphp/event-loop#232 and reactphp/stream#159

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,10 +18,10 @@
     },
     "require": {
         "php": ">=5.3",
-        "react/stream": "^1.0 || ^0.7"
+        "react/stream": "^1.2"
     },
     "require-dev": {
         "phpunit/phpunit": "^9.3 || ^5.7 || ^4.8",
-        "react/event-loop": "^1.0 || ^0.5 || ^0.4 || ^0.3"
+        "react/event-loop": "^1.2"
```
Comment (clue) [inline composer.json:25]: I don't think the updated examples reference the event loop at all?

**Review by clue — COMMENTED** [review]
@SimonFrings Thanks for looking into this!

File: composer.json (hunk 1 above)
Comment (SimonFrings) [reply composer.json:25]: The event loop is still necessary for some tests (see `FunctionalControlCodeParserTest.php`).

**Review by clue — APPROVED** [review]
@SimonFrings Thanks for the update, changes LGTM! :shipit:

**Pull request merged**